### ISSUE TITLE
[bug] broken malayalam font rendering

### ISSUE THREAD

**Issue by yozachar** [brave/brave-browser#34213]
## Description 
Malayalam rendering is broken.

## Steps to Reproduce

   1. Install brave browser
   2. Copy this text to the URL bar: `എന്നുവരും വ്യഞ്ജനം`

## Actual result:

![image](https://github.com/brave/brave-browser/assets/38415384/e2867f66-0413-4e95-a6b1-a590bf82520d)

## Expected result:

![image](https://github.com/brave/brave-browser/assets/38415384/5255a8c2-db25-4eca-b35e-98fb6b0fcefb)

## Reproduces how often: 

Everytime

## Brave version (brave://version info)

```r
Brave	          1.60.110 Chromium: 119.0.6045.105 (Official Build) unknown (64-bit) 
Revision	  0bb6e7a81aed340b5a9e8b752c1cf92768257c40
OS   	          Linux
JavaScript	  V8 11.9.169.6
User agent        Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36
Command Line      /app/brave/brave --disable-features=WebAssemblyTrapHandler,DesktopPWAsRunOnOsLogin --no-default-browser-check --disable-domain-reliability --enable-dom-distiller --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --variations-insecure-server-url=https://variations.brave.com/seed --disable-features=DesktopPWAsRunOnOsLogin,WebAssemblyTrapHandler --flag-switches-begin --flag-switches-end --component-updater=url-source=https://go-updater.brave.com/extensions
Executable Path   /app/brave/brave
Profile Path      /home/us-er/.var/app/com.brave.Browser/config/BraveSoftware/Brave-Browser/Default
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? `Yes`
- Can you reproduce this issue with the beta channel? `Probably`
- Can you reproduce this issue with the nightly channel? `I think so`

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? `No`
- Does the issue resolve itself when disabling Brave Rewards? `No`
- Is the issue reproducible on the latest version of Chrome? `Chromium, Yes`

## Miscellaneous Information:

```md
# System Details Report

## Report details
- **Date generated:**                              2023-11-09 08:14:21

## Hardware Information:
- **Hardware Model:**                              Lenovo ThinkBook 15 G5 ABP
- **Memory:**                                      24.0 GiB
- **Processor:**                                   AMD Ryzen™ 7 7730U with Radeon™ Graphics × 16
- **Graphics:**                                    AMD Radeon™ Graphics
- **Disk Capacity:**                               (null)

## Software Information:
- **Firmware Version:**                            LMCN16WW
- **OS Name:**                                     Arch Linux
- **OS Build:**                                    rolling
- **OS Type:**                                     64-bit
- **GNOME Version:**                               45.0
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.5.8-arch1-1
```

```console
$ LANG=ml_IN fc-match
Gayathri-Regular.otf: "Gayathri" "Regular"
$ cat ~/.config/fontconfig/conf.d/50-malayalam.conf
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
  <match>
    <test name="lang" compare="contains">
      <string>ml</string>
    </test>
    <test name="family">
      <string>sans-serif</string>
    </test>
    <edit name="family" mode="prepend">
      <string>Gayathri</string>
    </edit>
  </match>
  <alias>
    <family>Gayathri</family>
    <default>
      <family>sans-serif</family>
    </default>
  </alias>
</fontconfig>
```

- https://www.chromium.org/developers/linux-technical-faq/#system-integration
- https://github.com/brave/brave-browser/issues/12746 (2020)
- https://gitlab.com/api/v4/projects/6154882/jobs/artifacts/Version1.300/raw/build/gayathri-Version1.300.tar.gz?job=build-tag

**Comment by yozachar** [comment]
`terminus-font ttf-bitstream-vera ttf-croscore ttf-droid gnu-free-fonts ttf-ibm-plex ttf-liberation libertinus-font ttf-roboto tex-gyre-fonts ttf-ubuntu-font-family ttf-ms-fonts ttf-anonymous-pro ttf-cascadia-code ttf-fira-sans inter-font ttf-opensans adobe-source-sans-fonts otf-crimson gentium-plus-font libertinus-font tex-gyre-fonts ttf-croscore ttf-junicode xorg-fonts-type1 ttf-joypixels ttf-twemoji ttf-symbola ttf-arphic-uming ttf-indic-otf otf-latin-modern otf-latinmodern-math ttf-mac-fonts`

One of these packages was causing trouble. Removing these fixed the rendering issue.

**Comment by athulkrishna2015** [comment]
@yozachar thank you for your fix
how did you managed to identify /ttf-indic-otf as culprit

**Comment by yozachar** [comment]
> @yozachar thank you for your fix
> how did you managed to identify /ttf-indic-otf as culprit 
> 

Well, by trial and error, luckily that package was early on the list.

**Comment by chandujr** [comment]
I'm on Fedora and I don't have ttf-indic-otf installed. But still I'm seeing Malayalam rendering like this (Roboto font):

![Image](https://github.com/user-attachments/assets/0ce70aba-7235-4653-9910-9669ae5a78c7)

where it should have been like this (Firefox, Manjari font):

![Image](https://github.com/user-attachments/assets/a962779d-9c6f-4ff9-8acf-7c5fef55d202)

Anybody know why?

Brave version: 1.78.102 Chromium: 136.0.7103.113 (Official Build) unknown (64-bit) 

Turns out, [the issue is in the Flatpak version](https://github.com/brave/brave-browser/issues/46473#issuecomment-3139919153). I installed the native package and it is working fine.
